socialhome/content/models.py

```diff
@@ -118,6 +118,8 @@ def save(self, author=None, *args, **kwargs):
             # New with author, set a GUID and author
             self.guid = uuid4()
             self.author = author
+            if self.pinned:
+                self.order = Content.objects.filter(pinned=True).count() + 1
```

Comment:
Your query is wrong. We don't want the count of pinned content, we want the next unused value of `order`. So you need the maximum current value and then `+1`.